include/protocols/transaction-addon.h

```diff
@@ -0,0 +1,83 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef LABWC_PROTOCOLS_TRANSACTION_ADDON_H
+#define LABWC_PROTOCOLS_TRANSACTION_ADDON_H
+
+#include <wayland-server-core.h>
+
+struct lab_transaction {
+	uint32_t change;
+	void *src;
+	void *data;
+
+	struct {
+		struct wl_signal destroy;
+	} events;
+
+	// Private
+	struct wl_list link;
+};
+
+struct lab_transaction_session_context {
+	int ref_count;
+	struct wl_list transactions;
+};
+
+struct lab_wl_resource_addon {
+	struct lab_transaction_session_context *ctx;
+	void *data;
+};
+
+/*
+ * Creates a new addon which can be attached to a wl_resource via
+ * wl_resource_set_user_data() and retrieved via wl_resource_get_user_data().
+ *
+ * Usually the ctx argument should be addon->ctx of the parent wl_resource.
+ * If it is NULL it will be created automatically which can be used for top
+ * level wl_resources (when a client binds a wl_global from the registry).
+ *
+ * The context refcount is increased by one after this call.
+ */
+struct lab_wl_resource_addon *lab_resource_addon_create(
+	struct lab_transaction_session_context *ctx);
+
+/*
+ * A generic transaction attached to a session context list.
```

Comment:
I think this doesn't explain what `lab_transaction_add()` does.

Reply:
Suggestions appreciated, I am pretty bad at describing things.